lib/widgets/flame_sprite_widget.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+import '../sprite.dart';
+
+import 'dart:math';
+
+class FlameSpriteWidget extends StatelessWidget {
+  final Sprite sprite;
+  final bool center;
+
+  FlameSpriteWidget({
+    @required this.sprite,
+    this.center = false,
+  }) : assert(sprite.loaded(), 'Sprite must be loaded');
+
+  @override
+  Widget build(_) {
+    return Container(
+      child: CustomPaint(painter: _FlameSpritePainer(sprite, center)),
+    );
+  }
+}
+
+class _FlameSpritePainer extends CustomPainter {
+  final Sprite _sprite;
+  final bool _center;
+
+  _FlameSpritePainer(this._sprite, this._center);
+
+  @override
+  bool shouldRepaint(_FlameSpritePainer old) =>
+      old._sprite != _sprite || old._center != _center;
```

Comment:
let's do a body here, only use inline function if it actually fits in one line

Reply:
I strongly disagree with that, but I don't mind putting a body on he function.

Reply:
for me:

inline function iff. one liner
body function iff. multiple lines

but I am open to other formatting suggestions if you disagree :)

Reply:
I don't, I think that:

```dart
bool bla() =>
  expression1 ||
  expression2 ||
  expression3;
```

Than:

```dart
bool bla() {
  return expression1 ||
  expression2 ||
  expression3;
}
```

As they do exactly the same thing, but I can omite the `{}` and `return` on the first example

But don't worry, no need to discuss to much about this, as this is clearly a bikeshedding, I don't mind changing to using the body.

Reply:
The flutter style guide agrees with Luan:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using-for-short-functions-and-methods

Reply:
I guess I disagree with Flutter's styleguide on this one, but I already agreed on changing, don't worry, was just explaining my thoughts

Reply:
Yeah, tbh I think the inline one looks nicer when it is so short, but I think I like consistency more than that. :)

Reply:
Me too, that is why I don't argue too much about this, but I really would like that dartanalyzer could get this type of thing. I am a very strong believer that code style issue, should never be discussed on PRs, that should be the job of the automatic linter.

Hopefully dartanalyzer will be as good as eslint someday :)